app/src/test/java/com/dbychkov/data/datastore/DatabaseFlashcardDataStoreTest.java

```diff
@@ -0,0 +1,219 @@
+package com.dbychkov.data.datastore;
+
+import com.dbychkov.data.greendao.DaoSession;
+import com.dbychkov.data.greendao.FlashcardEntity;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+
+/**
+ * Created by brian on 5/6/16.
+ * Its very important to test the data access layer to make sure that CRUD ops are done correctly
+ * and data integrity is preserved
+ * This class tests the DatabaseFlashcardDataStore but DatabaseLessonDataStore,
+ * FlashcardDataRepository and LessonDataRepository should be tested in very similar way
+ */
+public class DatabaseFlashcardDataStoreTest {
+
+    DatabaseFlashcardDataStore databaseFlashcardDataStore;
+
+    /**
+     * init a databaseflashcarddatastore with injected mock object
+     *
+     * @throws Exception
+     */
+    @Before
+    public void setUp() throws Exception {
+        databaseFlashcardDataStore = new DatabaseFlashcardDataStore(mock(DaoSession.class));
+    }
+
+    /**
+     * init a new databaseflashcardatastore
+     *
+     * @throws Exception
+     */
+    @After
+    public void tearDown() throws Exception {
+        databaseFlashcardDataStore = new DatabaseFlashcardDataStore(mock(DaoSession.class));
```

Comment:
the goal was to de-allocated the memory of the databaseFlashcardDataStore that is being used in all of the tests. Now that I think about it, its completely redundant to implement the teardown method since setUp() already takes care of it at the start of every test. I think I was simply trying to be thorough without paying too much attention to the implications.